### DISCUSSION TITLE
How to customize option and command name colors in --help?

### DISCUSSION THREAD

**Discussion by malici0us-link** [fastapi/typer#660] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer
from typing import Annotated, Optional, List, Tuple
from rich import print as rprint

# Create Typer instance
app = typer.Typer(
    rich_markup_mode="rich",
    epilog="Made with [green]❤️[/green] on planet Earth.",
)

@app.callback()
def decorator_callback(
        authenticated: Annotated[bool, typer.Option(
            "[red]--auth[/red]", # This doesn't work — how do I change the option color?
            help="This is a [red]global top-level option[/red] defined in app.callback()."
        )] = False,
        verbose: Annotated[bool, typer.Option(
            help="This is [red]another global top-level option[/red] defined in app.callback().\n\n"
        )] = False,
):
    pass

if __name__ == "__main__":
    app()
```

### Description

How do you change the color of options and commands in --help? By default they are both a bright blue. Also, if you are using a boolean option, the negative boolean "--no-example" is purple. How can I change these bright blue and purple colors? I tried the code snippet I've attached here, but that doesn't seem to work. As you can see, in the code snippet, Rich syntax works for the help parameter (`[red]global top-level option[/red]` and `red]another global top-level option[/red]` work) and the epilog (the heart turns green), but it doesn't work when declaring the name of your option.

### Operating System

macOS

### Typer Version

0.9.0

### Python Version

3.11.4

### Additional Context

 
![Xnip2023-08-31_03-14-35PM](https://github.com/tiangolo/typer/assets/139306524/e6ecc60e-7eba-4073-b3c3-c0d090e8589c)

![Xnip2023-08-31_03-53-05PM](https://github.com/tiangolo/typer/assets/139306524/5d830be1-6c4c-4ece-be58-4593b6e10267)

**Comment by d3xma** [comment]
@malici0us-link any updates on this?
I would really like to have more control over the default colors of typer. @tiangolo any idea?

**Comment by Tremeschin** [comment]
@malici0us-link @dexma-dev

I wanted to change the color of the type hinting / ranges, as yellow was too intrusive, I did it doing the following:

```python
import typer.rich_utils

# Note: Must be `module.var = "Value"`, as `from typer.rich_utils import STYLE_METAVAR`
# copies the variable definition to this local file and doesn't change values upstream
typer.rich_utils.STYLE_METAVAR = "dim"

# (...)
```

There's a lot of color constants [defined there](https://github.com/tiangolo/typer/blob/04eba6b70203287176d2823753513226bf778872/typer/rich_utils.py#L30), go for some trial and error, should do it :)

**Reply by DavidWalshe93** [reply]
Thanks @Tremeschin 💯 This is ideal for helping end-users that suffer from color-impairment and cannot see the default styling very well :)

**Reply by Tremeschin** [reply]
@DavidWalshe93 That's an awesome use case we all should think more often! 🙂

I forgot to mention in the previous comment (a couple months old hehe) that one can find all [available colors](https://rich.readthedocs.io/en/stable/appendix/colors.html) and [styles](https://rich.readthedocs.io/en/stable/style.html) in the rich docs. I've also ended up using using the following style modifications on my projects:

```python
typer.rich_utils.STYLE_METAVAR = "italic grey42"
typer.rich_utils.STYLE_OPTIONS_PANEL_BORDER = "bold grey42"
typer.rich_utils.STYLE_OPTION_DEFAULT = "bold bright_black"
typer.rich_utils.DEFAULT_STRING = "(default: {})"
typer.rich_utils.STYLE_OPTIONS_TABLE_PADDING = (0, 1, 0, 0)
```

y'know, maybe someone could compile these custom stylings somewhere in one of those `awesome-typer-styles` type of repo 😆
